markdown/events/2022/bytesize-36-multiqc.md

```diff
@@ -24,3 +24,119 @@ It is our hope that these talks / videos will build an archive of training mater
+
+[22:33](https://www.youtube.com/watch?v=a3_IYSMrKAk&t=1353)
+You might also notice there's a couple of extra buttons up at the top here. That has been done with something called `--sample_names`, where you give MultiQC tab separated file with all your expected sample identifiers and then alternative sample names. The column head is then four buttons at the top. If I click, in this case "user supplied names", that's something custom I've labeled it, then you see all the sample names down there switch. We by default have NGI identifiers which is what's useful for us. But then our end users might not really know what that is. They can click that button and see all the sample names that they supplied to us really quickly, really easily. All that does is just pre-populate the MultiQC toolbox really quickly with lots of different sample matches. Dead easy to do and can be very, very helpful. Of course, this is an example of going to town with customizing your report output, just to give you a flavor for what's possible if you really, really go for it. This is a little Easter egg in MultiQC. See if you can take that out --template. Okay, I won't be too much longer. I'm running over a bit, sorry.
+
+[23:49](https://www.youtube.com/watch?v=a3_IYSMrKAk&t=1429)
+Looking to the future, a couple of things to look forward to with MultiQC. Those of you have heard me talk before might recognize some of these slides here. Most of this stuff has been planned for MultiQC since about 2018 or 19, which by coincidence is around the time that another one of my projects started taking off around that one is called nf-core and sucked up some of my time. Anyway, this is stuff which is being actively worked on and will happen. It's stuff I'm excited about. To kick us off is refactoring the code base, so that it works more as a Python library rather than purely a command line tool. Now if you want to, if you're using Jupyter notebooks or custom Python scripts, you can import MultiQC and you can run it on a this, in a programmatic way on a folder and it will generate the reports. What you can't do yet is generate a MultiQC reports object, and then pull out specific stats and specific plots on demand and that use all that internal functionality that's there. At the moment, that's a bit tricky, but I'm hoping to get there soon. It'll be a really useful interactive or script-based analysis tool as well as a command line tool.
```

Comment:
```suggestion
Looking to the future, a couple of things to look forward to with MultiQC. Those of you have heard me talk before might recognize some of these slides here. Most of this stuff has been planned for MultiQC since about 2018 or 19, which by coincidence is around the time that another one of my projects started taking off around that one is called nf-core and sucked up some of my time. Anyway, this is stuff which is being actively worked on and will happen. It's stuff I'm excited about. To kick us off is refactoring the code base, so that it works more as a Python library rather than purely a command line tool. Now if you want to, if you're using Jupyter notebooks or custom Python scripts, you can import MultiQC and you can run it like this, in a programmatic way on a folder and it will generate the reports. What you can't do yet is generate a MultiQC reports object, and then pull out specific stats and specific plots on demand and that use all that internal functionality that's there. At the moment, that's a bit tricky, but I'm hoping to get there soon. It'll be a really useful interactive or script-based analysis tool as well as a command line tool.
```

Reply:
track changes

> "[...] if you're using Jupyter notebooks or custom Python scripts, you can import MultiQC and you can run it ~~on a~~ **like** this, [...]"